evalml/tests/component_tests/test_time_series_featurizer.py

```diff
@@ -525,19 +589,100 @@ def test_delayed_feature_transformer_does_not_modify_input_data(delayed_features
     assert_frame_equal(X, expected)
 
 
+@pytest.mark.parametrize("rolling_window_size", [0.1, 0.2, 0.5, 0.75, 1.0])
+@pytest.mark.parametrize("gap", [0, 1, 2])
+@pytest.mark.parametrize("forecast_horizon", [1, 5, 10])
+@pytest.mark.parametrize("max_delay", [1, 3, 9])
+@patch(
+    f"evalml.pipelines.TimeSeriesFeaturizer.{DELAYED_FEATURES_METHOD_NAME}",
+)
+def test_time_series_featurizer_rolling_mean(
+    mock_delay,
+    max_delay,
+    forecast_horizon,
+    rolling_window_size,
+    gap,
+    delayed_features_data,
+):
+    X, y = delayed_features_data
+    mock_delay.return_value = X
+    output = TimeSeriesFeaturizer(
+        max_delay=max_delay,
+        forecast_horizon=forecast_horizon,
+        gap=gap,
+        rolling_window_size=rolling_window_size,
+        date_index="date",
+    ).fit_transform(X, y)
+
+    size = int(rolling_window_size * max_delay)
+    rolling_means = (
+        X.feature.shift(forecast_horizon + gap).rolling(size + 1, size + 1).mean()
+    )
+    rolling_means_target = (
+        y.shift(forecast_horizon + gap).rolling(size + 1, size + 1).mean()
+    )
+    expected = pd.DataFrame(
+        {
+            "date": X["date"],
+            "feature_rolling_mean": rolling_means,
+            "target_rolling_mean": rolling_means_target,
+        }
+    )
+    assert_frame_equal(output, expected)
+
+
+@pytest.mark.parametrize("rolling_window_size", [0.1, 0.2, 0.5, 0.75, 1.0])
+@pytest.mark.parametrize("gap", [0, 1, 2])
+@pytest.mark.parametrize("forecast_horizon", [1, 5, 10])
+@pytest.mark.parametrize("max_delay", [1, 3, 9])
+def test_time_series_featurizer_does_not_need_to_delay_to_compute_means(
+    max_delay, forecast_horizon, gap, rolling_window_size, delayed_features_data
+):
+    X, y = delayed_features_data
+    output = TimeSeriesFeaturizer(
+        max_delay=max_delay,
+        forecast_horizon=forecast_horizon,
+        gap=gap,
+        delay_features=False,
```

Comment:
Can this just be parameterized into the other test?  If you feel it's better tested out explicitly, that's cool too.  I assume that's why you did this!

Reply:
Yea I liked testing both halves of the feature engineering computation separately to make it easier to verify the output matched the expected value!